### PR TITLE
[READY] algorithms: tests: break up into multiple sets to help build time

### PR DESCRIPTION
**Pull request by fnrizzi**
this relates to #4450 
- test for std algorithms are broken into multiple sets 
- print statements in tests have been commented out for now

### CODE REVIEW

**Review by cz4rs — APPROVED** [review]
The split looks good :+1: 
For the print statements, this should probably follow whatever is decided in https://github.com/kokkos/kokkos/issues/4463

File: algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp [hunk 1]
```diff
@@ -329,10 +329,10 @@ void std_algorithms_min_max_element_test::test_minmax_element_non_trivial_data(
     }
     {
       const auto result =
-          KE::minmax_element(exespace(), KE::cbegin(view), KE::cend(view));
+          KE::minmax_element(exespace(), KE::begin(view), KE::end(view));
       std_algo_min_max_test_verify(gold, result.first, result.second, view);
       const auto result2 = KE::minmax_element("MYCUSTOMLABEL2", exespace(),
-                                              KE::cbegin(view), KE::cend(view));
+                                              KE::begin(view), KE::end(view));
```
Comment (crtrott) [inline algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:335]: Why this change?

File: algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp [hunk 2]
```diff
@@ -329,10 +329,10 @@ void std_algorithms_min_max_element_test::test_minmax_element_non_trivial_data(
     }
     {
       const auto result =
-          KE::minmax_element(exespace(), KE::cbegin(view), KE::cend(view));
+          KE::minmax_element(exespace(), KE::begin(view), KE::end(view));
```
Comment (crtrott) [inline algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:332]: Why this change?

**Review by crtrott — CHANGES_REQUESTED** [review]
One question regarding a change. Also did you guys test whether this speeds up things (i.e. link time)

File: algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp (hunk 2 above)
Comment (fnrizzi) [reply algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:332]: If we were to do:
```cpp
const auto result = KE::minmax_element(exespace(), KE::cbegin(view), KE::cend(view));
```
then result would be a pair of iterators computed as:
```cpp
result.first = cbegin() + something
result.second = cbegin() + something
```
So when we compare `result` against the gold solution , see line 206:
```cpp
EXPECT_EQ(result - KE::begin(testedView), goldPair.first);
```
since we use `begin` here, this raises an issue (at least on some compilers) because we are computing a distance from a const iterator and a non-const iterator. 

So by doing:
```cpp
const auto result = KE::minmax_element(exespace(), KE::begin(view), KE::end(view)); // notice we drop 'c'
```
then everything is consistent and works fine.

File: algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp (hunk 1 above)
Comment (fnrizzi) [reply algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:335]: same as above !

File: algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp (hunk 2 above)
Comment (fnrizzi) [reply algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:332]: i don't know why for this only some compilers complained, in fact in the CI this was not flagged. 
This only showed up in intel builds that Nathan pointed out I think,

**Comment by fnrizzi** [comment]
@crtrott 
- I did this also because you said you tried a build failed on Windows and it failed. I think this split should solve that. 
- this also helped a bit the build time of the tests. I tried locally on my machine and made a bit of a difference for even GCC/clang builds. I did not do it properly though. I have to do a proper check for this for OpenMPTarget which was one of the main reasons for doing this.

**Review by crtrott — APPROVED** [review]
